helios-services/src/main/java/com/spotify/helios/servicescommon/coordination/DefaultZooKeeperClient.java

```diff
@@ -108,23 +112,25 @@ public CuratorFramework getCuratorFramework() {
   @Override
   /** {@inheritDoc} */
   public void ensurePath(final String path) throws KeeperException {
-    assertClusterIdFlagTrue();
     ensurePath(path, false);
   }
 
   @Override
   /** {@inheritDoc} */
   public void ensurePath(final String path, final boolean excludingLast) throws KeeperException {
+    PathUtils.validatePath(path);
+
     assertClusterIdFlagTrue();
-    EnsurePath ensurePath = client.newNamespaceAwareEnsurePath(path);
-    if (excludingLast) {
-      ensurePath = ensurePath.excludingLast();
-    }
-    try {
-      ensurePath.ensure(client.getZookeeperClient());
-    } catch (Exception e) {
-      propagateIfInstanceOf(e, KeeperException.class);
-      throw propagate(e);
+
+    final String[] parts = path.substring(1).split(Pattern.quote("/"));
```

Comment:
It really isn't. I just always do `split(quote("string"))` out of an abundance of caution so that nobody comes back and tries to edit the string without realizing that it's a regex.